src/main/java/com/mraof/minestuck/item/crafting/alchemy/generator/GristCostGenerator.java

```diff
@@ -1,56 +1,59 @@
 package com.mraof.minestuck.item.crafting.alchemy.generator;
 
 import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.MinestuckConfig;
 import com.mraof.minestuck.item.crafting.alchemy.GristCostRecipe;
 import com.mraof.minestuck.item.crafting.alchemy.GristSet;
-import net.minecraft.server.packs.resources.ResourceManager;
-import net.minecraft.server.packs.resources.SimplePreparableReloadListener;
-import net.minecraft.util.profiling.ProfilerFiller;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.crafting.RecipeManager;
 import net.minecraftforge.event.AddReloadListenerEvent;
-import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.event.TagsUpdatedEvent;
+import net.minecraftforge.event.server.ServerStoppedEvent;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 import net.minecraftforge.fml.common.Mod;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import java.util.*;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 @Mod.EventBusSubscriber(modid = Minestuck.MOD_ID, bus = Mod.EventBusSubscriber.Bus.FORGE)
-public final class GristCostGenerator extends SimplePreparableReloadListener<Void>
+public final class GristCostGenerator
 {
 	private static final Logger LOGGER = LogManager.getLogger();
 	
-	private final RecipeManager recipes;
-	
-	private GristCostGenerator(RecipeManager recipes)
+	//Hacky access to the recipe manager, which isn't otherwise accessible in TagsUpdatedEvent
+	private static RecipeManager recipes;
+	@SubscribeEvent
+	public static void onResourceReload(AddReloadListenerEvent event)
 	{
-		this.recipes = recipes;
+		recipes = event.getServerResources().getRecipeManager();
 	}
-	
-	@SubscribeEvent(priority = EventPriority.LOW)
-	public static void addListener(AddReloadListenerEvent event)
+	@SubscribeEvent
+	public static void onServerStopped(ServerStoppedEvent event)
 	{
-		event.addListener(new GristCostGenerator(event.getServerResources().getRecipeManager()));
+		recipes = null;
 	}
 	
-	@Override
-	protected Void prepare(ResourceManager resourceManagerIn, ProfilerFiller profilerIn)
+	@SubscribeEvent
+	public static void onTagsUpdated(TagsUpdatedEvent event)
 	{
-		return null;
+		if(event.getUpdateCause() == TagsUpdatedEvent.UpdateCause.SERVER_DATA_LOAD)
+		{
+			if(recipes != null)
+				run(recipes);
+			else
+				LOGGER.error("Failed to access recipe manager for grist cost generation!");
+		}
```

Comment:
When rebasing your commit, it looks like you accidentally still included "changes" to GristCostGenerator, RecipeGeneratedCostProvider, and LandGatePlacement which essentially have reverted them to a past state. If you go to your branches in IntelliJ and left click on mraof/1.18, find the option that says "Show diff with working tree" and use that to fix these changes for those three files

Reply:
Will be handled in PR to KingBalDoro:1.18-items